### ISSUE TITLE
mismatching header files and libs causes seg fault

### ISSUE THREAD

**Issue by shi-yan** [JoeyDeVries/LearnOpenGL#376]
This project has a strange setup. 

it ships with assimp headers, but expect to use the prebuilt assimp binaries installed by the system. This will result in mismatched version of headers and libs. 

with the latest assimp 5 installed via brew on mac, I experienced a crash. My fix was removing the included assimp headers and only use the system install.

**Comment by JoeyDeVries** [comment]
Yeah, it's a bit of a weird situation. Pre-built binaries are supplied for windows users, but not for OS X and Linux users as they're it's preinstalled. Then for windows users it should include the headers as well, but then causes a mismatch on OSX/Linux. A PR was set up that fixes this on OSX/Linux, but breaks on Windows: https://github.com/JoeyDeVries/LearnOpenGL/pull/374

Maybe it's possible to set up CMake s.t. it won't use the shipped headers in case of OSX/Linux, but does support them on Windows, then it could work?

**Comment by shi-yan** [comment]
I think that works, or using cmake's external project to build everything from source?

https://cmake.org/cmake/help/latest/module/ExternalProject.html
